File: SBNZ/SBNZ/src/main/java/ftn/siit/sbnz/SBNZ/model/Query.java
package ftn.siit.sbnz.SBNZ.model;

import ftn.siit.sbnz.SBNZ.web.dto.QueryDTO;

public class Query {

    public enum Purpose {
        EVERYDAY, SPORT, OFFROAD, TRANSPORT
    }

    public enum Gear {
        MANUAL, AUTOMATIC, NA
    }

    private Integer budget;
    private Integer familySize;
    private Integer experience;
    private Purpose purpose;
    private Boolean regNotExpensive;
    private Boolean lowFuelConsumption;
    private Boolean lowKm;
    private Integer age;
    private Boolean luxury;
    private Car.Country country;
    private String make;
    private Gear gear;
    private Boolean environment;

    public Query() {
    }

    public Query(Integer budget, Integer familySize, Integer experience, Purpose purpose, Boolean regNotExpensive,
                 Boolean lowFuelConsumption, Integer age, Boolean luxury, Car.Country country, String make,
                 Gear gear, Boolean environment, Boolean lowKm) {
        this.budget = budget;
        this.familySize = familySize;
        this.experience = experience;
        this.purpose = purpose;
        this.regNotExpensive = regNotExpensive;
        this.lowFuelConsumption = lowFuelConsumption;
        this.age = age;
        this.luxury = luxury;
        this.country = country;
        this.make = make;
        this.gear = gear;
        this.environment = environment;
        this.lowKm = lowKm;
    }

    public Query(QueryDTO queryDTO) {
        if (queryDTO.getBudget() == null) {
            queryDTO.setBudget(0);
        }
        if (queryDTO.getFamilySize() == null) {
            queryDTO.setFamilySize(0);
        }
        if (queryDTO.getExperience() == null) {
            queryDTO.setExperience(0);
        }
        if (queryDTO.getAge() == null) {
            queryDTO.setAge(0);
        }
        this.budget = queryDTO.getBudget();
        this.familySize = queryDTO.getFamilySize();
        this.experience = queryDTO.getExperience();
        this.purpose = Purpose.valueOf(queryDTO.getPurpose());
        this.regNotExpensive = queryDTO.getRegNotExpensive();
        this.lowFuelConsumption = queryDTO.getLowFuelConsumption();
        this.age = queryDTO.getAge();
        this.luxury = queryDTO.getLuxury();
        this.country = Car.Country.valueOf(queryDTO.getCountry());
        this.make = queryDTO.getMake();
        this.gear = Gear.valueOf(queryDTO.getGear());
        this.environment = queryDTO.getEnvironment();
        this.lowKm = queryDTO.getLowKm();
    }

    public Integer getBudget() {
        return budget;
    }

    public void setBudget(Integer budget) {
        this.budget = budget;
    }

    public Integer getFamilySize() {
        return familySize;
    }

    public void setFamilySize(Integer familySize) {
        this.familySize = familySize;
    }

    public Integer getExperience() {
        return experience;
    }

    public void setExperience(Integer experience) {
        this.experience = experience;
    }

    public Purpose getPurpose() {
        return purpose;
    }

    public void setPurpose(Purpose purpose) {
        this.purpose = purpose;
    }

    public Integer getAge() {
        return age;
    }

    public void setAge(Integer age) {
        this.age = age;
    }

    public Boolean getLuxury() {
        return luxury;
    }

    public void setLuxury(Boolean luxury) {
        this.luxury = luxury;
    }

    public Car.Country getCountry() {
        return country;
    }

    public void setCountry(Car.Country country) {
        this.country = country;
    }

    public String getMake() {
        return make;
    }

    public void setMake(String make) {
        this.make = make;
    }

    public Gear getGear() {
        return gear;
    }

    public void setGear(Gear gear) {
        this.gear = gear;
    }

    public Boolean getEnvironment() {
        return environment;
    }

    public void setEnvironment(Boolean environment) {
        this.environment = environment;
    }

    public Boolean getRegNotExpensive() {
        return regNotExpensive;
    }

    public void setRegNotExpensive(Boolean regNotExpensive) {
        this.regNotExpensive = regNotExpensive;
    }

    public Boolean getLowFuelConsumption() {
        return lowFuelConsumption;
    }

    public void setLowFuelConsumption(Boolean lowFuelConsumption) {
        this.lowFuelConsumption = lowFuelConsumption;
    }

    public Boolean getLowKm() {
        return lowKm;
    }

    public void setLowKm(Boolean lowKm) {
        this.lowKm = lowKm;
    }
}

File: SBNZ/sbnzfrontend/src/components/cars/Query.js
import React from "react";
import {
  TextField,
  Select,
  Checkbox,
  MenuItem,
  Button,
  Container,
  Paper,
} from "@material-ui/core";
import Axios from "axios";
import FormTitle from "../layout/FormTitle";
import ViewOne from "../cars/ViewOne";

const Query = (props) => {
  const [budget, setBudget] = React.useState("");
  const [familySize, setFamilySize] = React.useState("");
  const [experience, setExperience] = React.useState("");
  const [purpose, setPurpose] = React.useState("EVERYDAY");
  const [purposeOptions] = React.useState(["EVERYDAY", "SPORT", "OFFROAD", "TRANSPORT"]);
  const [regNotExpensive, setRegNotExpensive] = React.useState(false);
  const [lowFuelConsumption, setLowFuelConsumption] = React.useState(false);
  const [lowKm, setLowKm] = React.useState(false);
  const [age, setAge] = React.useState("");
  const [luxury, setLuxury] = React.useState(false);
  const [country, setCountry] = React.useState("NA");
  const [countryOptions] = React.useState(["GERMANY", "ITALY", "FRANCE", "JAPAN", "USA", "RUSSIA", "ROMANIA", "CZECHIA", "SPAIN", "UK", "SOUTH_KOREA", "CHINA", "NA"]);
  const [make, setMake] = React.useState("");
  const [gear, setGear] = React.useState("NA");
  const [gearOptions] = React.useState(["MANUAL", "AUTOMATIC", "NA"]);
  const [environment, setEnvironment] = React.useState(false);
  const [cars, setCars] = React.useState([]);
  
  const handleCheckRegNotExpensive = (event) => {
    setRegNotExpensive(event.target.checked);
  };
  const handleCheckLowFuelConsumption = (event) => {
    setLowFuelConsumption(event.target.checked);
  };
  const handleCheckLowKm = (event) => {
    setLowKm(event.target.checked);
  };
  const handleCheckLuxury = (event) => {
    setLuxury(event.target.checked);
  };
  const handleCheckEnvironment = (event) => {
    setEnvironment(event.target.checked);
  };
  const handleSelectChangePurpose = (event) => {
    setPurpose(event.target.value);
  };
  const handleSelectChangeCountry = (event) => {
    setCountry(event.target.value);
  };
  const handleSelectChangeGear = (event) => {
    setGear(event.target.value);
  };
  const submitForm = () => {
    let query = {
      budget: budget,
      familySize: familySize,
      experience: experience,
      purpose: purpose,
      regNotExpensive: regNotExpensive,
      lowFuelConsumption: lowFuelConsumption,
      lowKm: lowKm,
      age: age,
      luxury: luxury,
      country: country,
      make: make,
      gear: gear,
      environment: environment,
    };
    console.log(query);
    Axios.post(
      "http://localhost:8080/query",
      query
    ).then((response) => setCars(response.data));
  };
  const purposeOptionsList = purposeOptions.map((opt) => (
    <MenuItem key={opt} value={opt}>
      {opt}
    </MenuItem>
  ));
  const countryOptionsList = countryOptions.map((opt) => (
    <MenuItem key={opt} value={opt}>
      {opt}
    </MenuItem>
  ));
  const gearOptionsList = gearOptions.map((opt) => (
    <MenuItem key={opt} value={opt}>
      {opt}
    </MenuItem>
  ));
  const viewCar = cars.map((c) => (
    <ViewOne car = {c} key = {c.id}/>
  ));
  return (
    <Container className="form-container">
      <Paper elevation={3}>
        <FormTitle>Enter your preferences</FormTitle>
        <form>
          <TextField
            type="number"
            label="Budget"
            value={budget}
            onChange={(event) => setBudget(event.target.value)}
          />
          <TextField
            type="number"
            label="Family Size"
            value={familySize}
            onChange={(event) => setFamilySize(event.target.value)}
          />
          <TextField
            type="number"
            label="Years of experience"
            value={experience}
            onChange={(event) => setExperience(event.target.value)}
          />
          <label>Purpose</label>
          <Select
            id="purpose"
            value={purpose}
            onChange={handleSelectChangePurpose}
          >
            {purposeOptionsList}
          </Select>
          <label>Registration not expensive</label>
          <Checkbox checked={regNotExpensive} onChange={handleCheckRegNotExpensive} />
          <label>Low fuel consumption</label>
          <Checkbox checked={lowFuelConsumption} onChange={handleCheckLowFuelConsumption} />
          <label>Low mileage</label>
          <Checkbox checked={lowKm} onChange={handleCheckLowKm} />
          <TextField
            type="number"
            label="Maximum car age"
            value={age}
            onChange={(event) => setAge(event.target.value)}
          />          
          <label>Luxury cars prefered</label>
          <Checkbox checked={luxury} onChange={handleCheckLuxury} />
          <label>Prefered country of origin</label>
          <Select
            id="country"
            value={country}
            onChange={handleSelectChangeCountry}
          >
            {countryOptionsList}
          </Select>
          <TextField
            label="Prefered make"
            value={make}
            onChange={(event) => setMake(event.target.value)}
          />
          <label>Prefered gear type</label>
          <Select
            id="gear"
            value={gear}
            onChange={handleSelectChangeGear}
          >
            {gearOptionsList}
          </Select>
          <label>Environment-friendly</label>
          <Checkbox checked={environment} onChange={handleCheckEnvironment} />

          <Button variant="contained" color="primary" onClick={submitForm}>
            Send
          </Button>
        </form>
      </Paper>
      {viewCar}
    </Container>
  );
};

export default Query;
